source/configuration_manual/mail_location/obox/obox_settings.rst

```diff
@@ -14,150 +14,243 @@ Enable obox plugin.
+   with ``10% +1d`` if the disk space used by index files of this priority
+   type take <= 10% of ``metacache_max_space``, increase the weight by
+   ``1d = 60*60*24 = 86400``.
+
+  * Because the initial weight is based on UNIX timestamp, the weight
+    adjustment is also given as time. This practically means that e.g.
+    ``+1d`` typically gives 1 extra day for the index files to exist
+    compared to index files that don't have the weight boost.
+  * ``<percentage>`` exists so that the weight boost doesn't cause some
+    index files to dominate too much. For example, if root indexes' weights
+    weren't limited, it could be possible that the system would be full of
+    only root indexes and active users' other indexes would be cleaned
+    almost immediately.
+
+The ``metacache_size_weights`` setting is used to do final adjustments
+depending on the disk space used by this user's indexes of the specific
+priority. The setting is in format
+``<low size> <low weight adjustment> <max size> <high weight adjustment>``.
+
+The weight adjustment calculation is:
+
+ * If disk space is equal or less than ``<low size>``, increase weight by
+   ``(<low size> - <disk space>) * <low weight adjustment> / <low size>``
+ * Otherwise, cap the ``<disk space>`` to ``<max size>`` and increase weight
+   by ``(<disk space> - <low size>) * <high weight adjustment> / (<max size> - <low size>)``
  * The idea here is to give extra weight boost for
-   * small indexes, because they're small enough that it won't matter if they live longer than most and
-   * very large indexes, because it's so expensive to keep uploading/downloading them in object storage
- * With the default "2M +30 1G +120" value the priority adjustments will look like:
-   * 0 kB: +30
-   * 500 kB: +23
-   * 1 MB: +15
-   * 1,5 MB: +8
-   * 2 MB: 0
-   * 10 MB: +1
-   * 50 MB: +6
-   * 100 MB: +12
-   * 258 MB: +30
-   * 500 MB: +60
-   * >=1 GB: +120
-
-The rest of the obox settings are specific to the object storage backend that is used. 
-
-It can be useful to flush unimportant changes in metacache every night when the system has idle capacity. This way if users are moved between backends, there's somewhat less work to do on the new backends since caches are more up-to-date. This can be done by running "doveadm metacache flushall" in a cronjob.
+
+  * Small indexes, because they're small enough that it won't matter if they
+    live longer than most, AND
+  * Very large indexes, because it's so expensive to keep
+    uploading/downloading them in object storage
+
+ * With the default ``2M +30 1G +120`` value the priority adjustments will
+   look like:
+
+  * 0 kB: ``+30``
+  * 500 kB: ``+23``
+  * 1 MB: ``+15``
+  * 1,5 MB: ``+8``
+  * 2 MB: ``0``
+  * 10 MB: ``+1``
+  * 50 MB: ``+6``
+  * 100 MB: ``+12``
+  * 258 MB: ``+30``
+  * 500 MB: ``+60``
+  * >=1 GB: ``+120``
+
+The rest of the obox settings are specific to the object storage backend that
+is used. 
```

Comment:
Maybe we can move that to the non-advanced part and add links to `:ref:obox_drivers` and  `:ref:`http_based_object_storages` ?

Reply:
HTTP based object storages page is related to specific obox drivers.  So at the moment, I don't think a link should be added to this page (it should be added from each driver that uses HTTP. And maybe a generic link from the drivers page).